engine/src/engine.rs

```diff
@@ -201,6 +200,30 @@ impl From<BalanceOverflow> for GasPaymentError {
     }
 }
 
+pub struct ERC20Address(Address);
+impl AsRef<[u8]> for ERC20Address {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+impl From<Vec<u8>> for ERC20Address {
+    fn from(bytes: Vec<u8>) -> Self {
+        Self(Address::from_slice(&bytes))
+    }
+}
+
+pub struct NEP141Account(AccountId);
+impl AsRef<[u8]> for NEP141Account {
+    fn as_ref(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+}
+impl From<Vec<u8>> for NEP141Account {
+    fn from(bytes: Vec<u8>) -> Self {
+        Self(AccountId::try_from(bytes).unwrap())
+    }
+}
```

Comment:
I do think we should keep these as a try_from and then catch them in the logic. We could in the future use this logic elsewhere where it may fail.

Reply:
Done in https://github.com/aurora-is-near/aurora-engine/pull/342/commits/339dc43def5853776090c5ff17d65f8d788bbd3a